### DISCUSSION TITLE
Extend expression language

### DISCUSSION THREAD

**Discussion by Silvanoc** [linkml/linkml#2401] [Ideas]
Current [expression language](https://linkml.io/linkml/schemas/expression-language.html) has poor support for handling lists (only the equivalent to a python list comprehension like following is supported: `[item.field for item in items]`). It would be useful to have more advanced possibilities.

The language is declarative and should remain like that. Python list comprehensions are also declarative, but too bound to python. A more programming-language independent specification would be desirable.

_jq_ could be taken as an expression language inspiration/reference (as proposed by @cmungall) or even simply taken over... we could then use the available implementation. Since instances can be serialized as JSON and _jq_ can be [integrated into python](https://github.com/mwilliamson/jq.py), why not directly extend it that way?

**Comment by Ostrzyciel** [comment]
I definitely support this. Even something as simple as `ifabsent` already uses some sort of an under-specified language (see https://github.com/linkml/linkml/issues/3834). This is a major barrier to implementing runtimes and generators for languages that support expressions/inference.
